db/migrate/20150702160751_add_user_id_to_articles.rb

```diff
@@ -0,0 +1,40 @@
+class AddUserIdToArticles < ActiveRecord::Migration
+  def up
+    #add user id column
+    add_column :articles, :user_id, :integer
+
+    #converts username to id
+    execute <<-SQL
+      UPDATE articles
+      SET user_id = users.id
+      FROM users
+      WHERE users.username = articles.author
+    SQL
+
+    #drop author column from article
+    remove_column :articles, :author
+
+    #add null constraint to user id
+    change_column_null :articles, :user_id, false
+  end
+
+  def down
+    #add author column
+    add_column :articles, :author, :string
+
+    #converts id to username
+    execute <<-SQL
+      UPDATE articles
+      SET author = users.username
+      FROM users
+      WHERE users.id = articles.user_id
+    SQL
+
+    #drop user id column
+    remove_column :articles, :user_id
+
+    #drop null constrint to user id
```

Comment:
I think you might need to update this comment -- seems like you'd want to re-add the null constraint on `author` rather than remove the constraint for `user_id`.